src/index.js

```diff
@@ -1,54 +1,145 @@
+/**
+ * Query github for semantic version tag.
+ *
+ * @typedef {{tag: string, version: string, hash: string, date: Date}} Tag
+ */
+
 'use strict';
 
 const semver = require('semver');
 const childProcess = require('child-process-promise');
 
 const tagRegex = /tag:\s*([^,)]+)/g;
-const commitDetailsRegex = /.*?(tag:\s*([^,)]+).*);(.+);(.+)/;
+const commitDetailsRegex = /^(.+);(.+);(.+)$/;
 
+/**
+ * Run shell command and resolve with stdout content
+ *
+ * @param  {string} command Shell command
+ * @return {Promise<string,Error>}
+ */
 function runCommand(command) {
   return childProcess.exec(command)
     .then(result => result.stdout);
 }
 
-function getFirstValidTag(allTags, guess) {
-  let match;
-  let result = guess;
-  while (match = tagRegex.exec(allTags)) {
-    if (semver.valid(match[1])) {
-      result = match[1];
-      break;
+/**
+ * Get all tags with a semantic version name out of a list of Refs
+ *
+ * @param  {string} refs List of refs
+ * @return {Arrays<Tag>}
+ */
+function getSemanticTags(refs) {
+  const tags = [];
+
+  while (true) {
+    const match = tagRegex.exec(refs);
+
+    if (match == null) {
+      return tags;
     }
-  }
 
-  return result;
+    const tag = match[1];
+    const version = semver.valid(tag);
+
+    if (version != null) {
+      tags.push({ tag, version, hash: null, date: null });
+    }
+  }
 }
 
 function isString(v) {
   return typeof v === 'string';
 }
 
+/**
+ * Parse commit into an array of tag object.
+ *
+ * @param  {string} line Line to parse
+ * @return {Array<Tag>}
+ */
+function parseLine(line) {
+  const match = commitDetailsRegex.exec(line);
+
+  if (!match || match.length < 4) {
+    return [];
+  }
+
+  const tags = getSemanticTags(match[1]);
```

Comment:
Just nitpicking: `getSemanticTags` is returning the list of tags and versions from a particular commit, so how about renaming it to something like `extractCommitTagsAndVersions` ?

`const tagsAndVersions = extractCommitTagsAndVersions(match[1]);` ?!